### ISSUE TITLE
bigdl-llm-init does not work similar to bigdl-nano-init for torch cpp_extension

### ISSUE THREAD

**Issue by Uxito-Ada** [intel-analytics/BigDL#9367]
I encounter the problem when replacing `source bigdl-nano-init` with `source bigdl-llm-init` [here](https://github.com/Uxito-Ada/BigDL/blob/main/python/llm/dev/benchmark/all-in-one/run-deepspeed-spr.sh#L2):

```bash
Traceback (most recent call last):
  File "/home/shaojun/BigDL/python/llm/dev/benchmark/all-in-one/run.py", line 645, in <module>
    run_model(model, api, conf['in_out_pairs'], conf['local_model_hub'], conf['warm_up'], conf['num_trials'], conf['num_beams'], conf['low_bit'])
  File "/home/shaojun/BigDL/python/llm/dev/benchmark/all-in-one/run.py", line 60, in run_model
    result = run_deepspeed_llama2(repo_id, local_model_hub, in_out_pairs, warm_up, num_trials, 1)
  File "/home/shaojun/BigDL/python/llm/dev/benchmark/all-in-one/run.py", line 430, in run_deepspeed_llama2
    model = deepspeed.init_inference(model, mp_size=world_size,
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/__init__.py", line 342, in init_inference
    engine = InferenceEngine(model, config=ds_inference_config)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/inference/engine.py", line 125, in __init__
    self._create_model_parallel_group(config)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/inference/engine.py", line 250, in _create_model_parallel_group
    init_distributed()
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/comm/comm.py", line 635, in init_distributed
    init_deepspeed_backend(get_accelerator().communication_backend_name(), timeout, init_method)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/comm/comm.py", line 160, in init_deepspeed_backend
    ccl_backend = CCLBackend(rank=rank, world_size=size, timeout=timeout, init_method=init_method)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/comm/ccl.py", line 27, in __init__
    self.ccl_comm_op = build_ccl_op()
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/comm/ccl.py", line 19, in build_ccl_op
    ccl_cpp_module = builder.load()
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/ops/op_builder/builder.py", line 446, in load
    return self.jit_load(verbose)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/deepspeed/ops/op_builder/builder.py", line 489, in jit_load
    op_module = load(name=self.name,
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1308, in load
    return _jit_compile(
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1736, in _jit_compile
    return _import_module_from_library(name, build_directory, is_python_module)
  File "/root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 2136, in _import_module_from_library
    module = importlib.util.module_from_spec(spec)
  File "<frozen importlib._bootstrap>", line 565, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1173, in create_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
ImportError: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/.cache/torch_extensions/py39_cpu/deepspeed_ccl_comm/deepspeed_ccl_comm.so)
```

It seems that there is any set of lib path missing in bigdl-llm-init compared with bigdl-nano-init.

The output prints of the two tools are as following:

1. bigdl-nano-init
```bash
(ziteng-llm) [root@spr01 all-in-one]# source bigdl-nano-init
Sourcing bigdl-nano-init in: /root/anaconda3/envs/ziteng-llm/bin
Setting OMP_NUM_THREADS...
Setting KMP_AFFINITY...
Setting KMP_BLOCKTIME...
Setting jemalloc...
nano_vars.sh already exists
+++++ Env Variables +++++
LD_PRELOAD            = /root/anaconda3/envs/ziteng-llm/lib/libiomp5.so /root/anaconda3/envs/ziteng-llm/lib/python3.9/site-packages/bigdl/nano/libs/libjemalloc.so
MALLOC_CONF           = oversize_threshold:1,background_thread:false,metadata_thp:always,dirty_decay_ms:-1,muzzy_decay_ms:-1
OMP_NUM_THREADS       = 96
OMP_PROC_BIND         =
OMP_SCHEDULE          =
MKL_NUM_THREADS       =
MKL_DYNAMIC           =
KMP_AFFINITY          = granularity=fine,none
KMP_BLOCKTIME         = 1
TF_ENABLE_ONEDNN_OPTS = 1
+++++++++++++++++++++++++
Complete.
```

2. bigdl-llm-init
```bash
(ziteng-llm) [root@spr01 all-in-one]# source bigdl-llm-init
found intel-openmp in /root/anaconda3/envs/ziteng-llm/lib/libiomp5.so
+++++ Env Variables +++++
LD_PRELOAD            = /root/anaconda3/envs/ziteng-llm/lib/libiomp5.so
OMP_NUM_THREADS       = 48
+++++++++++++++++++++++++
Complete.
```

It is shown that `LD_PRELOAD` and `MALLOC_CONF` ENVs of the two are different, while I manually set llm's to be similar as nano's does not solve the problem.

**Comment by qiyuangong** [comment]
```bash
ImportError: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/.cache/torch_extensions/py39_cpu/deepspeed_ccl_comm/deepspeed_ccl_comm.so)
```

This line told us`deepspeed_ccl_comm.so` used a higher version glibc than current env.

Try to link host glibc to conda env
https://askubuntu.com/questions/1418016/glibcxx-3-4-30-not-found-in-conda-environment

**Comment by Uxito-Ada** [comment]
Fixed in #9419
